core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -5366,22 +5366,25 @@ public void setPreparedStatementValue( DatabaseMeta databaseMeta, PreparedStatem
             if ( getLength() == DatabaseMeta.CLOB_LENGTH ) {
               setLength( databaseMeta.getMaxTextFieldLength() );
             }
-            String string = getString( data );
-            int len = string.length();
-            int maxlen = isLengthInvalidOrZero() ? len : getLength();
-            if ( len <= maxlen ) {
-              preparedStatement.setString( index, string );
+
+            if ( getLength() <= databaseMeta.getMaxTextFieldLength() ) {
```

Comment:
This needs to be `<=` since it is being set to `databaseMeta.getMaxTextFieldLength()` a few lines before.